### PR TITLE
Unchecked array access leading to potential corruption

### LINKED ISSUE #131: Unchecked array access leading to potential corruption
Related to #56
From the FIXME in `src/fheroes2/kingdom/kingdom.cpp`.

`kingdomscount` is not checked before use. If `kingsdomcount > KINGDOMMAX` this will write data in memory and will lead to undefined behavior.

### PR DESCRIPTION
**Pull request by adrianca88**
Check array size before write it to avoid memory corruption
relates to #131

### CODE REVIEW

File: src/fheroes2/kingdom/kingdom.cpp [hunk 1]
```diff
@@ -809,11 +809,13 @@ StreamBase & operator<< (StreamBase & msg, const Kingdoms & obj)
...
 StreamBase & operator>> (StreamBase & msg, Kingdoms & obj)
 {
-    u32 kingdomscount;
-    msg >> kingdomscount; // FIXME: check kingdomscount
+    u32 kingdomscount = 0;
+    msg >> kingdomscount;
 
-    for(u32 ii = 0; ii < kingdomscount; ++ii)
-	msg >> obj.kingdoms[ii];
+    if (kingdomscount < KINGDOMMAX) {
```
Comment (ihhub) [inline src/fheroes2/kingdom/kingdom.cpp]: How about `kingdomscount <= KINGDOMMAX`?

Comment (adrianca88) [reply src/fheroes2/kingdom/kingdom.cpp]: In my opinion it is more correct (see my commit fefce6c). What do you think?

Comment (ihhub) [reply src/fheroes2/kingdom/kingdom.cpp]: `#define KINGDOMMAX			6` which means that the number of kingdoms more than 6 then it's a bug while equal should be correct.

Comment (ihhub) [reply src/fheroes2/kingdom/kingdom.cpp]: Yes, you are right.

**Comment by ihhub** [comment]
@adrianca88 thank you very much for the contribution to the project!

**Pull request merged**